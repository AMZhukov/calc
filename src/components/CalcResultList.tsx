import React from 'react';

type CalcResultListProps = {
    resultList: string[],
}

export const CalcResultList: React.FC<CalcResultListProps> = ({ resultList }) => {
  return (
    <ul className="calc__result-list" data-testid="resultList">
      {resultList.map((item, index) => {
        return (
          // eslint-disable-next-line react/no-array-index-key
          <li className="calc__result-item" id={`result-${index}`} key={index}>{item}</li>
        );
      })}
    </ul>
  );
};
