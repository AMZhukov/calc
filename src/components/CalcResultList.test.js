import { render, screen } from '@testing-library/react';
import { CalcResultList } from './CalcResultList';
import React from 'react';
import '@testing-library/jest-dom/extend-expect'

test('render Calc result list', () => {
    render(<CalcResultList resultList={['1 + 1 = 2', '3 / 2 = 1,5', '10 - 2 = 8']} />);

    const resultList = screen.getByTestId('resultList');
    expect(resultList).toBeInTheDocument();
    expect(resultList).toMatchSnapshot();
})
