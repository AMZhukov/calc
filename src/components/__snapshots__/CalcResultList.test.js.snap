// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Calc result list 1`] = `
<ul
  class="calc__result-list"
  data-testid="resultList"
>
  <li
    class="calc__result-item"
    id="result-0"
  >
    1 + 1 = 2
  </li>
  <li
    class="calc__result-item"
    id="result-1"
  >
    3 / 2 = 1,5
  </li>
  <li
    class="calc__result-item"
    id="result-2"
  >
    10 - 2 = 8
  </li>
</ul>
`;
